castai/types.go

```diff
@@ -75,6 +75,28 @@ func (c *ClusterAction) Data() interface{} {
 	return nil
 }
 
+func (c *ClusterAction) NodeID() string {
+	if c.ActionDeleteNode != nil {
+		return c.ActionDeleteNode.NodeID
+	}
+	if c.ActionDrainNode != nil {
+		return c.ActionDrainNode.NodeID
+	}
+	if c.ActionPatchNode != nil {
+		return c.ActionPatchNode.NodeID
+	}
+	if c.ActionApproveCSR != nil {
+		return c.ActionApproveCSR.NodeID
+	}
+	if c.ActionCheckNodeDeleted != nil {
+		return c.ActionCheckNodeDeleted.NodeID
+	}
+	if c.ActionCheckNodeStatus != nil {
+		return c.ActionCheckNodeStatus.NodeID
+	}
+	return "nil"
```

Comment:
From action API perspective it looks like NodeID is optional (for example it is irrelevant for AddNode action). Then return value should be pointer. It means that there should IFs where node id is used for logging.

Reply:
Yeah that makes sense. I have removed this part completely as having node ID in the handleAction and ackAction methods looks excessive.